client/src/project/shared/Projects.state.js

```diff
@@ -103,24 +103,48 @@ class ProjectsCoordinator {
     });
   }
 
+  _setLandingProjects(projectList, lastVisited) {
+    this.model.setObject({
+      landingProjects: {
+        fetched: new Date(),
+        fetching: false,
+        list: { $set: projectList },
+        lastVisited,
+      }
+    });
+  }
 
   async getLanding() {
     if (this.model.get("landingProjects.fetching"))
       return;
     // set status to fetching, get the projects for the landing page
     this.model.set("landingProjects.fetching", true);
-    const params = { order_by: "last_activity_at", per_page: 5, membership: true };
     try {
-      const landingProjects = await this.client.getProjects({ ...params });
-      const projectList = landingProjects?.data?.map((project) => this._starredProjectMetadata(project)) ?? [];
-      this.model.setObject({
-        landingProjects: {
-          fetched: new Date(),
-          fetching: false,
-          list: { $set: projectList },
-        }
-      });
-      return { landing: projectList };
+      const lastProjects = await this.client.getRecentProjects(4);
+      const lastProjectsVisited = lastProjects?.data?.projects;
+      let projectList = [];
+      if (lastProjectsVisited?.length > 0) {
+        const projectRequests = [];
+        for (const project of lastProjectsVisited)
+          projectRequests.push(this.client.getProject(project));
+
+        Promise.allSettled(projectRequests).then( results => {
+          for (const result of results) {
+            if (result?.status === "fulfilled" && result?.value?.data?.all)
+              projectList.push(this._starredProjectMetadata(result?.value?.data.all));
+          }
+          this._setLandingProjects(projectList, true);
+        }).catch( () => {
+          this.model.set("landingProjects.fetching", false);
+        });
+      }
+      else {
```

Comment:
```suggestion
      if (!lastProjectsVisited?.length || !projectList.length) {
```

It's possible the projects previously visited have been deleted or otherwise not available (permissions, ...).
It's a rare case, but the message would be confusing for someone with some projects (see screenshot).
I wouldn't put too much effort into solving this case in a very elegant way.
Checking if at least one valid project was fetched from the list should be enough (the "recent projects" still flashes for a second or two, but it's a rare case, I think it's fine)

![Screenshot_20220225_101857](https://user-images.githubusercontent.com/43481553/155689182-c80fe1ff-12d7-48d4-b2cc-6a677aa4a09e.png)